pkg/reconciler/trigger/trigger.go

```diff
@@ -0,0 +1,424 @@
+	brokerLister       listers.BrokerLister
+	serviceLister      corev1listers.ServiceLister
+	tracker            tracker.Interface
+}
+
+// Check that our Reconciler implements controller.Reconciler.
+var _ controller.Reconciler = (*Reconciler)(nil)
+
+// NewController initializes the controller and is called by the generated code.
+// Registers event handlers to enqueue events.
+func NewController(
+	opt reconciler.Options,
+	triggerInformer eventinginformers.TriggerInformer,
+	channelInformer eventinginformers.ChannelInformer,
+	subscriptionInformer eventinginformers.SubscriptionInformer,
+	brokerInformer eventinginformers.BrokerInformer,
+	serviceInformer corev1informers.ServiceInformer,
+) *controller.Impl {
+
+	r := &Reconciler{
+		Base:               reconciler.NewBase(opt, controllerAgentName),
+		triggerLister:      triggerInformer.Lister(),
+		channelLister:      channelInformer.Lister(),
+		subscriptionLister: subscriptionInformer.Lister(),
+		brokerLister:       brokerInformer.Lister(),
+		serviceLister:      serviceInformer.Lister(),
+	}
+	impl := controller.NewImpl(r, r.Logger, ReconcilerName, reconciler.MustNewStatsReporter(ReconcilerName, r.Logger))
+
+	r.Logger.Info("Setting up event handlers")
+	triggerInformer.Informer().AddEventHandler(reconciler.Handler(impl.Enqueue))
+
+	// Tracker is used to notify us that a Trigger's Broker has changed so that
+	// we can reconcile.
+	r.tracker = tracker.New(impl.EnqueueKey, opt.GetTrackerLease())
+	brokerInformer.Informer().AddEventHandler(reconciler.Handler(
+		// Call the tracker's OnChanged method, but we've seen the objects
+		// coming through this path missing TypeMeta, so ensure it is properly
+		// populated.
+		controller.EnsureTypeMeta(
+			r.tracker.OnChanged,
+			v1alpha1.SchemeGroupVersion.WithKind("Broker"),
+		),
+	))
+
+	subscriptionInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: controller.Filter(v1alpha1.SchemeGroupVersion.WithKind("Trigger")),
+		Handler:    reconciler.Handler(impl.EnqueueControllerOf),
+	})
+	return impl
+}
+
+// Reconcile compares the actual state with the desired, and attempts to
+// converge the two. It then updates the Status block of the Trigger resource
+// with the current status of the resource.
+func (r *Reconciler) Reconcile(ctx context.Context, key string) error {
+	ctx = logging.WithLogger(ctx, r.Logger.Desugar().With(zap.String("key", key)))
+
+	// Convert the namespace/name string into a distinct namespace and name.
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		r.Logger.Errorf("invalid resource key: %s", key)
+		return nil
+	}
+
+	// Get the Trigger resource with this namespace/name.
+	original, err := r.triggerLister.Triggers(namespace).Get(name)
+	if apierrs.IsNotFound(err) {
+		// The resource may no longer exist, in which case we stop processing.
+		logging.FromContext(ctx).Error("trigger key in work queue no longer exists", zap.Any("key", key))
```

Comment:
Don't include the 'key'. That is automatic because you saved it into the context on line 134.

Reply:
done.